src/main/java/com/wavefront/sdk/jaxrs/client/ClientSpanDecorator.java

```diff
@@ -0,0 +1,81 @@
+package com.wavefront.sdk.jaxrs.client;
+
+import io.opentracing.Span;
+
+import java.net.MalformedURLException;
+
+import io.opentracing.tag.Tags;
+
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientResponseContext;
+
+import static com.wavefront.sdk.jaxrs.Constants.WF_SPAN_HEADER;
+
+/**
+ * Decorators for span tags and operation name.
+ *
+ * @author Hao Song (songhao@vmware.com).
+ */
+public interface ClientSpanDecorator {
+
+  /**
+   * Decorate get by incoming object.
+   */
+  void decorateRequest(ClientRequestContext requestContext, Span span);
+
+  /**
+   * Decorate spans by outgoing object.
+   */
+  void decorateResponse(ClientResponseContext responseContext, Span span);
+
+  /**
+   * Adds standard tags:
+   * {@link io.opentracing.tag.Tags#SPAN_KIND}, {@link io.opentracing.tag.Tags#COMPONENT},
+   * {@link io.opentracing.tag.Tags#HTTP_METHOD}, {@link io.opentracing.tag.Tags#HTTP_URL},
+   * {@link io.opentracing.tag.Tags#HTTP_STATUS}, {@link io.opentracing.tag.Tags#ERROR}
+   */
+  ClientSpanDecorator STANDARD_TAGS = new ClientSpanDecorator() {
+    @Override
+    public void decorateRequest(ClientRequestContext requestContext, Span span) {
+      Tags.COMPONENT.set(span, "jaxrs-client");
+      Tags.HTTP_METHOD.set(span, requestContext.getMethod());
+      try {
+        Tags.HTTP_URL.set(span, requestContext.getUri().toURL().toString());
+      } catch (MalformedURLException e) {
+        // ignoring returning null
+      }
+    }
+
+    @Override
+    public void decorateResponse(ClientResponseContext responseContext, Span span) {
+      Tags.HTTP_STATUS.set(span, responseContext.getStatus());
+      int statusCode = responseContext.getStatus();
+      if (statusCode >= 400 && statusCode <= 599) {
+        Tags.ERROR.set(span, true);
+      }
+    }
+  };
+
+  /**
+   * Set operation of the span using the server-side path injected by Wavefront Filter.
+   * For non-instrumented server side, the operation name will be just HTTP method name.
+   */
+  ClientSpanDecorator WF_PATH_OPERATION_NAME = new ClientSpanDecorator() {
+    String methodName = "";
+
+    @Override
+    public void decorateRequest(ClientRequestContext clientRequestContext, Span span) {
+      this.methodName = clientRequestContext.getMethod();
```

Comment:
This isn't thread-safe when processing multiple requests. You should use ThreadLocal instead.